src/satisfaction.js

```diff
@@ -62,7 +62,7 @@ export const getSatisfactionRatingsData = async function (req) {
 
     var issueSatisfaction = [];
     for (var issue of data.issues) {
-        if ( issue.fields[customFieldID].mySatisfactionRating ) {
+        if ( issue.fields[customFieldID] && issue.fields[customFieldID].mySatisfactionRating ) {
```

Comment:
I think you should add a check in this loop to ensure that the issue is part of the current project. I noticed that if you create another project board within your dev instance, those issues in the other board also appear in this API call so doing a check with the project ID can prevent getting and using data from another board. You can copy the code in `Line 58` and `Line 67 `in `src/selfPerformance.js` to do the check. Should probably do the same when getting motivation data @andrwcao @jiexiangfan.

Reply:
Done, thanks for picking this up, would never have realised this myself hahaha